### PR TITLE
make groups great again

### PR DESCRIPTION
**Pull request by JulianWielga**
simplified grouping behavior (from 4 buttons to 2) with selection
fixed minor bugs (name in properties)
changed custom link default icon

### CODE REVIEW

File: ui/client/reducers/graph/utils.ts [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
+  const validConnections = getSelectedNodes(state)
+    .map(({id}) => edges
+      .filter(isConnectedTo(id))
+      .filter(isConnectedTo(...without(selectionState, id))))
+    .map(validEgdes => validEgdes.length)
+
+  const lonelyNodes = validConnections.filter(l => l === 0).length
+  const edgeNodes = validConnections.filter(l => l === 1).length
+
+  return !lonelyNodes && edgeNodes === 2
```
Comment (mproch) [inline ui/client/reducers/graph/utils.ts:45]: https://staging.nussknacker.io/visualization/Bounded113?businessView=false - I think `edgeNodes === 2` condition does not allow such group?

File: ui/client/cypress/fixtures/testProcess2.json [hunk 2]
```diff
@@ -0,0 +1,287 @@
+{
```
Comment (mproch) [inline ui/client/cypress/fixtures/testProcess2.json:1]: I think we'd better have simpler, smaller diagram here (e.g. using mostly built-in nodes like filter). Now it's unnecessarily brittle - if we change implementation of our demo nodes, this test will break, as the process will be no longer valid

File: ui/client/components/toolbars/group/buttons/UngroupButton.tsx [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
+export function UngroupButton(): JSX.Element {
+  const graph = useSelector(getGraph)
   const {t} = useTranslation()
   const dispatch = useDispatch()
 
   return (
-    <CapabilitiesToolbarButton
-      write
-      name={t("panels.actions.group-ungroup.button", "ungroup")}
+    <ToolbarButton
```
Comment (mproch) [inline ui/client/components/toolbars/group/buttons/UngroupButton.tsx:16]: Why it's no longer Capabilities based?

File: ui/client/components/toolbars/group/buttons/GroupButton.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import {getGraph} from "../../../../reducers/selectors/graph"
+import ToolbarButton from "../../../toolbarComponents/ToolbarButton"
+
+export function GroupButton(): JSX.Element {
+  const graph = useSelector(getGraph)
+  const {t} = useTranslation()
+  const dispatch = useDispatch()
+
+  return (
+    <ToolbarButton
```
Comment (mproch) [inline ui/client/components/toolbars/group/buttons/GroupButton.tsx:16]: Why it's no longer Capabilities based?

File: ui/client/reducers/graph/utils.ts [hunk 5]
```diff
@@ -1,30 +1,51 @@
...
-      ...state,
-      groupingState: canGroup(state, node) ?
-        state.groupingState.concat(node.id) :
-        state.groupingState,
-    }
+export function getSelectedGroups(state: GraphState): GroupType[] {
+  return getSelectedNodes(state).filter(NodeUtils.nodeIsGroup)
+}
+
+export function canGroupSelection(state: GraphState): boolean {
```
Comment (mproch) [inline ui/client/reducers/graph/utils.ts:24]: Maybe some unit test would be nice for this?

**Review by mproch — COMMENTED** [review]
Nice changes, I added some comments

File: ui/client/components/toolbars/group/buttons/GroupButton.tsx (hunk 4 above)
Comment (lciolecki) [reply ui/client/components/toolbars/group/buttons/GroupButton.tsx:16]: I agree with @mproch - why you removed it?

**Review by lciolecki — APPROVED** [review]
Nice change - please let's look for my comments!

File: ui/client/components/toolbars/group/buttons/GroupButton.tsx (hunk 4 above)
Comment (JulianWielga) [reply ui/client/components/toolbars/group/buttons/GroupButton.tsx:16]: no reason. mistake with merge.

File: ui/client/components/toolbars/group/buttons/UngroupButton.tsx (hunk 3 above)
Comment (JulianWielga) [reply ui/client/components/toolbars/group/buttons/UngroupButton.tsx:16]: no reason. mistake with merge.

File: ui/client/components/tips/NodeErrorsLinkSection.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  const {nodeIds, message, showDetails, currentProcess, className} = props
+  const name = useSelector(getProcessUnsavedNewName)
+
+  return !isEmpty(nodeIds) && (
+    <div className={className}>
+      <ErrorHeader message={message}/>
+      {
+        nodeIds.map((nodeId, index) => {
+          const details = nodeId === "properties" ?
+            NodeUtils.getProcessProperties(currentProcess, name) :
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/components/tips/NodeErrorsLinkSection.tsx:27]: it's not only name. currentProcess comes from parents, not straight from store.

File: ui/client/reducers/graph/utils.ts (hunk 5 above)
Comment (JulianWielga) [reply ui/client/reducers/graph/utils.ts:24]: coverage from cypress, do you really need unit?

Zrzut ekranu 2021-05-5 o 12 53 47

Comment (mproch) [reply ui/client/reducers/graph/utils.ts:24]: maybe not *that* really, but I think cypress coverage is misleading in this case - lines are ok, but I don't think current tests capture all cases...

**Review by mproch — APPROVED** [review]
:+1:

**Pull request merged**